packages/itwinui-react/src/core/Tile/Tile.tsx

```diff
@@ -137,154 +173,185 @@ export type TileProps = {
-        isSelected={isSelected}
-        isNew={isNew}
-        status={status}
-      />
+    const [isMenuVisible, setIsMenuVisible] = React.useState(false);
+    const showMenu = React.useCallback(() => setIsMenuVisible(true), []);
+    const hideMenu = React.useCallback(() => setIsMenuVisible(false), []);
+    const [localActionable, setLocalActionable] =
+      React.useState(isActionableProp);
 
-      <span className='iui-tile-name-label'>{name}</span>
-    </div>
-  );
+    const isActionable = isActionableProp ?? localActionable;
 
-  return (
-    <div
-      className={cx(
-        'iui-tile',
-        {
-          'iui-folder': variant === 'folder',
-          'iui-new': isNew,
-          'iui-selected': isSelected,
-          'iui-actionable': isActionable,
-          [`iui-${status}`]: !!status,
-          'iui-loading': isLoading,
-        },
-        className,
-      )}
-      aria-disabled={isDisabled}
-      tabIndex={isActionable && !isDisabled ? 0 : undefined}
-      {...rest}
-    >
-      {variant !== 'folder' ? tileName : null}
+    const tileName = (
+      <div className='iui-tile-name'>
+        <TitleIcon
+          isLoading={isLoading}
+          isSelected={isSelected}
+          isNew={isNew}
+          status={status}
+        />
 
-      {thumbnail && (
-        <div className='iui-tile-thumbnail'>
-          {typeof thumbnail === 'string' ? (
-            <div
-              className='iui-tile-thumbnail-picture'
-              style={{ backgroundImage: `url(${thumbnail})` }}
-            />
-          ) : thumbnail && (thumbnail as JSX.Element).type === 'img' ? (
-            React.cloneElement(thumbnail as JSX.Element, {
-              className: 'iui-tile-thumbnail-picture',
-            })
-          ) : React.isValidElement(thumbnail) ? (
-            React.cloneElement(thumbnail, {
-              className: cx('iui-thumbnail-icon', thumbnail.props.className),
-            })
+        <span className='iui-tile-name-label'>
+          {isActionable && onClick ? (
+            <LinkOverlay as='button'>{name}</LinkOverlay>
           ) : (
-            thumbnail
+            name
           )}
+        </span>
+      </div>
+    );
 
-          {leftIcon &&
-            React.cloneElement(leftIcon as React.ReactElement, {
-              className: 'iui-tile-thumbnail-type-indicator',
-              'data-iui-size': 'small',
-            })}
+    return (
+      <TileContext.Provider value={{ setActionable: setLocalActionable }}>
+        <div
+          className={cx(
+            'iui-tile',
+            {
+              'iui-folder': variant === 'folder',
+              'iui-new': isNew,
+              'iui-selected': isSelected,
+              'iui-actionable': isActionable,
+              [`iui-${status}`]: !!status,
+              'iui-loading': isLoading,
+            },
+            className,
+          )}
+          aria-disabled={isDisabled}
+          onClick={onClick}
```

Comment:
don't need this onClick anymore

Reply:
This onClick is "div" event and I don't want to make LinkOverlay as div.

Reply:
i dont think there is much difference in a div event vs button event.

you might able to change the type of `onClick` to something like `React.MouseEventHandler<HTMLElement>`